spec/models/provider_enrichment_spec.rb

```diff
@@ -96,10 +96,81 @@
   end
 
   describe 'validation' do
+    describe 'on update' do
+      let(:provider) { build(:provider) }
+      let(:existing_provider_code) { nil }
+      let(:enrichment) { create(:provider_enrichment, provider: provider, provider_code: existing_provider_code) }
+
+      describe 'email' do
+        it 'validates email is present' do
+          enrichment.email = ""
+          enrichment.valid?
+
+          expect(enrichment.errors[:email]).to include("^Enter an email address in the correct format, like name@example.com")
+        end
+
+        it 'validates email contains an @ symbol' do
+          enrichment.email = "meow"
+          enrichment.valid?
+
+          expect(enrichment.errors[:email]).to include("^Enter an email address in the correct format, like name@example.com")
+        end
+
+        it 'does not validate if the email was not updated' do
+          enrichment.website = "cats4lyf.cat"
+          expect(enrichment.valid?).to be true
+        end
+      end
+
+      describe 'telephone' do
+        let(:valid_telephone_numbers) do
+          [
+            "+447123 123 123",
+            "+447123123123",
+            "07123123123"
```

Comment:
extension numbers

brackets...

Reply:
Added in tests for extension numbers and brackets :tada:

Reply:
```sql
SELECT DISTINCT( masked )
FROM   (SELECT Regexp_replace(t, '[0-9]', 'n', 'g') AS masked,
               t
        FROM   (SELECT DISTINCT( json_data ->> 'Telephone' ) AS t
                FROM   provider_enrichment) AS b
        WHERE  t IS NOT NULL
               AND t ~ '[^0-9 ]') AS m
ORDER  BY masked  
```
can we have a bit more test please, there is about 16 patterns

```
"masked"
"(****) *** ****"
"(*****) ******"
"(+**) (*)**** ******"
"+** (*) *** **** ***"
"+**(*)*** ** *****"
"*** **** **** ext ***"
"**** *** ****  ext ***"
"**** *** **** ext ****"
"***** ****** ext ***"
"***** ****** ext. ***" # this one has a dot
"***** ****** ext***"
"*********** ext ***"
"*** **** **** x***"
"**** *** **** x***"
"***** ****** x***"
"*********** x***"

```

Reply:
:+1: will do

Reply:
```
            # "(1234) 123 1234",
            # "1234 123 1234 x123",
            # "1234 123 1234 ext 1234",
            # "1234 123 1234  ext 123",
            # "+44(0)123 12 12345",
```

Looks like these are the formats that are flagged as invalid with this current implementation - thoughts?

Reply:
Does anything in our current data match these? Will they get flagged as invalid?
I prefer to be as permissive as we can

Reply:
Yeah @tvararu and I have been thinking the same - will look at just doing some basic checking for things like letters (outside of `ext`/`x`/brackets)